iam-k8s.tf

```diff
@@ -1,19 +1,36 @@
 resource "aws_iam_role" "waggle_dance_k8s_role_iam" {
-  count = var.wd_instance_type == "k8s" && var.waggle_dance_glue_assume_role_policy != "" ? 1 : 0
+  count = var.wd_instance_type == "k8s" && var.oidc_provider != "" ? 1 : 0
 
   name = "${local.instance_alias}-k8s-${var.aws_region}"
   tags = var.tags
 
   description = "Role to allow AWS Glue access from WaggleDance"
 
-  assume_role_policy = var.waggle_dance_glue_assume_role_policy
-
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
```

Comment:
Shouldn't this be multiple for example if I am in account X and I get federated to from account Y and Z how would I allow them?

Reply:
Actually, this is the part I am a little confused about, even after reading about [Glue cross-account access](https://docs.aws.amazon.com/glue/latest/dg/cross-account-access.html#cross-account-how-works) several times.

My assumption is that this will work sort of like S3 objects and federation:

- The `datalake` component that creates a Schema in Glue will need to somehow create either 
   -  An IAM role/policy that allows the WD role to assume this role and get access, or
   - A Glue resource policy that allows the WD role to read the Glue resources.
- The `federation` component/Waggledance just needs the permissions to read the Glue resources in the other account, which is what this PR sets up.

But I am fuzzy about how all this will work.  @rpoluri , any insight here?

Reply:
this part is for services running in k8s to use IAM using IRSA standard, so just one is sufficient... after there service(waggle-dance) will access glue in other account using iam policy(not assume policy)